WordPress/Classes/ViewRelated/NewGutenberg/NewGutenbergViewController.swift

```diff
@@ -521,18 +607,77 @@ extension NewGutenbergViewController {
         present(lightboxVC, animated: true)
     }
 
-    // TODO: reimplement
-//    func gutenbergDidRequestMention(callback: @escaping (Swift.Result<String, NSError>) -> Void) {
-//        DispatchQueue.main.async(execute: { [weak self] in
-//            self?.showSuggestions(type: .mention, callback: callback)
-//        })
-//    }
-//
-//    func gutenbergDidRequestXpost(callback: @escaping (Swift.Result<String, NSError>) -> Void) {
-//        DispatchQueue.main.async(execute: { [weak self] in
-//            self?.showSuggestions(type: .xpost, callback: callback)
-//        })
-//    }
+}
+
+// MARK: - Suggestions implementation
+
+extension NewGutenbergViewController {
+
+    private func showSuggestions(type: SuggestionType, callback: @escaping (Swift.Result<String, NSError>) -> Void) {
+        // Prevent multiple suggestions UI instances - simply ignore if already showing
+        guard currentSuggestionsController == nil else {
+            return
+        }
+        guard let siteID = post.blog.dotComID else {
+            callback(.failure(GutenbergSuggestionsViewController.SuggestionError.notAvailable as NSError))
+            return
+        }
+
+        switch type {
+        case .mention:
+            guard SuggestionService.shared.shouldShowSuggestions(for: post.blog) else { return }
+        case .xpost:
+            guard SiteSuggestionService.shared.shouldShowSuggestions(for: post.blog) else { return }
+        }
+
+        previousFirstResponder = view.findFirstResponder()
+        let suggestionsController = GutenbergSuggestionsViewController(siteID: siteID, suggestionType: type)
+        currentSuggestionsController = suggestionsController
+        suggestionsController.onCompletion = { (result) in
+            callback(result)
+            // Clear the current controller reference
+            self.currentSuggestionsController = nil
+            self.suggestionViewBottomConstraint = nil
+            // Clean up the UI
+            suggestionsController.view.removeFromSuperview()
+            suggestionsController.removeFromParent()
+            if let previousFirstResponder = self.previousFirstResponder {
```

Comment:
I think you can just use this variable without storing it in the view controller.